### PR TITLE
[#139551335] Jobs that run too often

### PR DESCRIPTION
**Pull request by Gathondu**
#### What does this PR do?
Add a status for jobs that run too often
#### Description of Task to be completed?
Add status and notification for jobs running too often

### CODE REVIEW

File: hc/api/management/commands/sendalerts.py [hunk 1]
```diff
@@ -19,10 +19,13 @@ def handle_many(self):
         query = Check.objects.filter(user__isnull=False).select_related("user")
 
         now = timezone.now()
-        going_down = query.filter(alert_after__lt=now, status="up")
+        going_down = query.filter(alert_after__lt=now, status = "up")
```
Comment (osule) [inline hc/api/management/commands/sendalerts.py]: Catch this space

File: hc/api/models.py [hunk 2]
```diff
@@ -87,6 +89,8 @@ def get_status(self):
         now = timezone.now()
 
         if self.last_ping + self.timeout + self.grace > now:
+            if self.status == "often":
```
Comment (osule) [inline hc/api/models.py:92]: This doesn't need to be nested. We really don't require to check this condition`self.last_ping + self.timeout + self.grace > now` to return a status for _often_ checks

File: hc/api/views.py [hunk 3]
```diff
@@ -20,11 +20,14 @@ def ping(request, code):
...
         return HttpResponseBadRequest()
-
+    
     check.n_pings = F("n_pings") + 1
     check.last_ping = timezone.now()
     if check.status in ("new", "paused"):
         check.status = "up"
+    if check.alert_after and check.last_ping < check.alert_after - check.grace-check.grace:
+        check.status = "often"
+        check.alert_often_pings = True
```
Comment (osule) [inline hc/api/views.py:30]: Why do we require _alert_often_pings_ together with status to know if a check alerts often?

**Review by osule — CHANGES_REQUESTED** [review]

**Comment by NgangaMaryanne** [comment]
making reviews

File: hc/api/views.py (hunk 3 above)
Comment (NgangaMaryanne) [reply hc/api/views.py:30]: The alert_often_pings is what stops the function sendalerts from sending too many alerts when a check is in often mode. without it, the alerts come every second. We were setting it to 'False' after sending one alert.

**Review by osule — APPROVED** [review]
LGTM

**Pull request merged**